tools/utils.py

```diff
@@ -104,6 +106,20 @@ def __cmp(self, other: 'Version', comparator: T.Callable[[T.Any, T.Any], bool])
         # versions are equal, so compare revisions
         return comparator(self._r, other._r)
 
+def wrap_path(name: str) -> Path:
+    return Path('subprojects', f'{name}.wrap')
+
+def read_wrap(name: str) -> configparser.ConfigParser:
+    config = configparser.ConfigParser(interpolation=None)
+    config.read(wrap_path(name), encoding='utf-8')
+    return config
+
+def write_wrap(path: Path, config: configparser.ConfigParser) -> None:
+    # configparser write() adds multiple trailing newlines, collapse them
+    buf = io.StringIO()
+    config.write(buf)
+    path.write_text(buf.getvalue().rstrip('\n') + '\n', encoding='utf-8')
```

Comment:
pathlib concrete methods such as write_path() or resolve() tend to be rather scary and full of weird edge cases. The old version in the other file(s) used `with open(...) as f: f.write()` for this sort of thing, and I think that pattern is quite helpful.

It's also faster and even manages to be more readable as it avoids mixing encoding arguments with the string you're building for the contents.